CMakeLists.txt

```diff
@@ -1,283 +1,83 @@
-	endif()
-endif()
-
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/Modules/")
-set(ENABLE_SCRIPTING TRUE CACHE BOOL "Enables scripting")
-set(SCRIPTING_ENABLED OFF CACHE BOOL "Internal global cmake variable" FORCE)
-
+# Use target folders for MSVC/Xcode/etc.
+include(DeprecationHelpers)
 include(ObsHelpers)
-include(ObsCpack)
-include(GNUInstallDirs)
-include(CheckCCompilerFlag)
-include(CheckCXXCompilerFlag)
-
-# Must be a string in the format of "x.x.x-rcx"
-if(DEFINED RELEASE_CANDIDATE)
-	set(OBS_VERSION "${RELEASE_CANDIDATE}")
-	string(REPLACE "-rc" "." RC_SPLIT ${RELEASE_CANDIDATE})
-	string(REPLACE "." ";" RC_SPLIT ${RC_SPLIT})
-	message(WARNING "******************************************************************************\nRelease candidate deteced, OBS_VERSION is now: ${OBS_VERSION}\n******************************************************************************")
-	list(GET RC_SPLIT 0 OBS_RELEASE_CANDIDATE_MAJOR)
-	list(GET RC_SPLIT 1 OBS_RELEASE_CANDIDATE_MINOR)
-	list(GET RC_SPLIT 2 OBS_RELEASE_CANDIDATE_PATCH)
-	list(GET RC_SPLIT 3 OBS_RELEASE_CANDIDATE)
-else()
-	set(OBS_RELEASE_CANDIDATE_MAJOR 0)
-	set(OBS_RELEASE_CANDIDATE_MINOR 0)
-	set(OBS_RELEASE_CANDIDATE_PATCH 0)
-	set(OBS_RELEASE_CANDIDATE 0)
-endif()
-
-# Binary Versioning for Windows
-if(WIN32)
-	string(REPLACE "-" ";" UI_VERSION_SPLIT ${OBS_VERSION})
-	list(GET UI_VERSION_SPLIT 0 UI_VERSION)
-	string(REPLACE "." ";" UI_VERSION_SEMANTIC ${UI_VERSION})
-	list(GET UI_VERSION_SEMANTIC 0 UI_VERSION_MAJOR)
-	list(GET UI_VERSION_SEMANTIC 1 UI_VERSION_MINOR)
-	list(GET UI_VERSION_SEMANTIC 2 UI_VERSION_PATCH)
-	set(OBS_COMPANY_NAME "OBS")
-	set(OBS_PRODUCT_NAME "OBS Studio")
-	set(OBS_COMMENTS "Free and open source software for video recording and live streaming")
-	set(OBS_LEGAL_COPYRIGHT "(C) Hugh Bailey")
-	set(MODULE_DESCRIPTION "OBS Studio")
-	configure_file(UI/obs.rc.in ${PROJECT_BINARY_DIR}/obs.rc)
-endif()
-
-if(MSVC AND NOT EXISTS "${CMAKE_BINARY_DIR}/ALL_BUILD.vcxproj.user")
-	file(GENERATE
-		OUTPUT "${CMAKE_BINARY_DIR}/ALL_BUILD.vcxproj.user"
-		INPUT "${CMAKE_SOURCE_DIR}/cmake/ALL_BUILD.vcxproj.user.in")
+# Set default compiler flags
+include(CompilerConfig)
+
+# Prohibit in-source builds
+file(TO_CMAKE_PATH "${PROJECT_BINARY_DIR}/CMakeLists.txt" _LOC_PATH)
+if(EXISTS "${LOC_PATH}")
+	message(FATAL_ERROR "OBS: You cannot build in a source directory (or any directory with "
+						"CMakeLists.txt file). Please make a build subdirectory. Feel free to "
+						"remove CMakeCache.txt and CMakeFiles.")
 endif()
+unset(_LOC_PATH)
 
+# Allow selection of common build types via UI
 if(NOT CMAKE_BUILD_TYPE)
-        set(CMAKE_BUILD_TYPE RelWithDebInfo)
+	set(CMAKE_BUILD_TYPE "RelWithDebInfo" CACHE STRING
+		"OBS build type [Release, RelWithDebInfo, Debug, MinSizeRel]" FORCE)
+	set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS Release RelWithDebInfo Debug MinSizeRel)
 endif()
 
-set(CMAKE_CXX_STANDARD 17)
-set(CMAKE_CXX_STANDARD_REQUIRED YES)
```

Comment:
Following up on [this comment](https://github.com/obsproject/obs-studio/pull/4560#issuecomment-828177203) and the removal of `CMAKE_CXX_STANDARD` and `CMAKE_CXX_STANDARD_REQUIRED` from CMakeLists.txt, Qt's updated [CMake documentation for Qt 6](https://doc.qt.io/qt-6/cmake-get-started.html) seems to recommend including them:
> Qt 6 requires a compiler supporting C++ version 17 or newer. Enforcing this by setting the CMAKE_CXX_STANDARD, CMAKE_CXX_STANDARD_REQUIRED variables will let CMake print an error if the compiler is too old.

Reply:
As long as the standard is set for the targets using Qt. It looks we set the standard for the UI to cpp 14. https://github.com/obsproject/obs-studio/pull/4560/commits/e62ccc1b28b673386cff247bbd15fb0515f7e687#diff-8018418025a9ba789bb1f6380af11291f2c5960de567c79575ca8f8b4223b65aR233

So maybe we want to set it to 17 instead?

Reply:
Yep sounds good, added it to all targets that either link to `obs-frontend-api` or include any Qt library directly. Will probably commit the patch tomorrow.